### PR TITLE
Homework8: Alexey Kozhan

### PR DESCRIPTION
**Pull request by AlexKozhan**
homework8 is done in 3 python files

### CODE REVIEW

File: homework1.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    return False
+
+
+solution1 = [1, 2, 3]
+solution2 = [1, 2, 1, 2]
+solution3 = [1, 3, 2, 1]
+solution4 = [1, 2, 3, 4, 5, 3, 5, 6]
+solution5 = [40, 50, 60, 10, 20, 30]
+
+print(strictly_increasing_without_one_element(solution1))
```
[the remark this answers is not in the document]
Comment (AlexKozhan) [reply homework1.py]: > pls, use assert instead of print()

fixed

File: homework1.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    return False
+
+
+solution1 = [1, 2, 3]
+solution2 = [1, 2, 1, 2]
+solution3 = [1, 3, 2, 1]
+solution4 = [1, 2, 3, 4, 5, 3, 5, 6]
+solution5 = [40, 50, 60, 10, 20, 30]
+
+assert strictly_increasing_without_one_element(solution1)
```
Comment (Stml89) [inline homework1.py:55]: what does this `assert` verify?

use of assert is following:
`assert <statement> <logical operatos> <expected value>, <error message>`
e.g.:
`assert strictly_increasing_without_one_element(solutions) == True, <error message>`

https://www.w3schools.com/python/ref_keyword_assert.asp

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework1.py (hunk 2 above)
Comment (AlexKozhan) [reply homework1.py:55]: > what does this `assert` verify?
> 
> use of assert is following: `assert <statement> <logical operatos> <expected value>, <error message>` e.g.: `assert strictly_increasing_without_one_element(solutions) == True, <error message>`
> 
> https://www.w3schools.com/python/ref_keyword_assert.asp

i did like this, but there were linter errors with asserting to True,
 E712 Avoid equality comparisons to `True`; use `if strictly_increasing_without_one_element(solution1):` for truth checks

Comment (Stml89) [reply homework1.py:55]: you are right, then do following:
```
assert <statement>, <error message>
```

**Review by Stml89 — COMMENTED** [review]
fix comment and you can merge PR

**Review by Stml89 — APPROVED** [review]

**Comment by Stml89** [comment]
to be able to merge it you have to resolve conflicts first, ask for help if you need it

**Pull request merged**